metricbeat/module/rabbitmq/exchange/_meta/data.json

```diff
@@ -0,0 +1,45 @@
+{
+    "@timestamp": "2017-10-12T08:05:34.853Z",
+    "beat": {
+        "hostname": "host.example.com",
+        "name": "host.example.com"
+    },
+    "metricset": {
+        "host": "localhost:15672",
+        "module": "rabbitmq",
+        "name": "exchange",
+        "rtt": 115
+    },
+    "rabbitmq": {
+        "exchange": {
+            "arguments": {},
+            "auto_delete": false,
+            "durable": true,
+            "internal": false,
+            "messages": {
+                "ack": {},
```

Comment:
Normally in case of empty objects they should not be added. Perhaps an issue in the schema? Not a blocker of the PR but worth mentioning.

Reply:
I haven't found how to avoid it, they are definned as optional `schema.Object`s, there would be other way to define them?
```
                "messages": c.Dict("message_stats", s.Schema{
                        "publish": s.Object{
                                "count": c.Int("publish", s.Optional),
                                "details": c.Dict("publish_details", s.Schema{
                                        "rate": c.Float("rate"),
                                }, c.DictOptional),
                        },
                        ...
                }
```

Reply:
@ruflin would it be ok to define the schema with dotted field names?
```
                "messages": c.Dict("message_stats", s.Schema{
                        "publish.count": c.Int("publish", s.Optional),
                        "publish.details": c.Dict("publish_details", s.Schema{
                                "rate": c.Float("rate"),
                        }, c.DictOptional),
```

Reply:
I wonder if this will have any affect on the output in the data.json?

For the empty objects: Can you try to set the surrounding doc as optional too? Just had a look at `ack` and I think it's not set to optional.